sdk/storage/Azure.Storage.Blobs/src/PartitionedUploader.cs

```diff
@@ -163,6 +163,13 @@ internal class PartitionedUploader
             {
                 scope.Start();
 
+                // Wrap progressHandler in a AggregatingProgressIncrementer to prevent
+                // progress from being reset with each stage blob operation.
+                if (progressHandler != null)
+                {
+                    progressHandler = new AggregatingProgressIncrementer(progressHandler);
```

Comment:
In the parallel upload case, we end up having nested `AggregatingProgressIncrementer` on the `ProgressIncrementingStream`, since we are first creating an `AggregatingProgressIncrementer` here, and then creating a new one in `BlockBlobClient.StageBlockInternal`. This causes the numbers reported back to the customer `IProgress` handler to be too high. The number should be the total number of bytes, otherwise customers wouldn't know what progress percentage to show to users. This is what seems to be happening:

The `ProgressIncrementingStream` reports back the number of bytes read to the first `AggregatingProgressIncrementer`. This incrementer then reports back the same number to its own inner `AggregatingProgressIncrementer`, which gets reported to the `IProgress` handler. However, on subsequent reads the initial incrementer will incorrectly increment its inner incrementer, because it reports back its `Current` property which is the total current progress, not the offset from the current read. So on each read, we are increasing the progress not by the additional bytes read, but by the total bytes read so far for each stream. You can see this if you run a test that will hit the parallel upload code path -  has to either be file size over 256 MB or set a lower minimum for the `StorageTransferOptions.MaximumTransferLength`. I think the way to fix this is to make sure that we don't nest the AggregatingProgressIncrementers by only creating one if needed in the `WithProgress` extension method.

Reply:
I updated StorageProgressExtensions.WithProgress() to ensure we don't nest AggregatingProgressIncrementer.